workflows/stairway.py

```diff
@@ -55,53 +57,74 @@ def ts_to_stairway(self, ts_path, num_bootstraps=1, mask_file=None):
-            # print(ts.get_num_mutations())
-            # print("This is the number of variants")
-            # print(ts.variants())
-            # print("This is the number of sites")
-            # print(ts.num_sites)
-            #retain = np.full(ts.get_num_mutations(), False)
-
-            # This part is not really working (dimension problem)
-            # Masking
-            retain = np.full(ts.num_sites, False)
-            # if mask_file:
-            #     mask_table = pd.read_csv(mask_file, sep="\t", header=None)
-            #     chrom = ts_p.split("/")[-1].split(".")[0]
-            #     sub = mask_table[mask_table[0] == chrom]
-            #     mask_ints = pd.IntervalIndex.from_arrays(sub[1], sub[2])
-            #     snp_locs = [int(x.site.position) for x in ts.variants()]
-            #     tmp_bool = [mask_ints.contains(x) for x in snp_locs]
-            #     retain = np.logical_or(retain, tmp_bool)
-            #     total_length -= np.sum(mask_ints.length)
-
-            #retain = retain.flatten()
-            retain = np.logical_not(retain)
-            # print("This is the shape of retain")
-            # print(retain.shape)
-            # append unmasked SFS
-            SFSs.append(allel.sfs(allel.HaplotypeArray(haps).count_alleles()[:, 1])[1:])
-            # get masked allele counts and append SFS
-            allele_counts = allel.HaplotypeArray(haps[retain,:]).count_alleles()
-            #allele_counts = allel.HaplotypeArray(haps).count_alleles()
-            SFSs.append(allel.sfs(allele_counts[:, 1])[1:])
-            sfs_path = ts_p+".sfs.pdf"
-            plots.plot_sfs(SFSs, sfs_path)
-            # Bootstrap allele counts
+
+            # Mapping mutation type IDs to class of mutation (e.g., neutral, non-neutral)
+            class_muts = {}
+            for dfe in ts.metadata["stdpopsim"]["DFEs"]:
+                for mt in dfe["mutation_types"]:
+                    mid = mt["slim_mutation_type_id"]
+                    if not mid in class_muts:
+                        class_muts[mid] = "neutral" if mt["is_neutral"] else "non_neutral"
+
+            site_class = np.empty(ts.num_sites, dtype=object)
+
+            # Finding neutral positions
+            neu_positions = []
+            non_neu_positions = []
+            for j, s in enumerate(ts.sites()):
+                mut_hits = []
+                for m in s.mutations:
+                    for md in m.metadata["mutation_list"]:
+                        mut_hits.append(md["mutation_type"])
+                        if set(class_muts[md["mutation_type"]]) == set("neutral"):
+                            neu_positions.append(m.site)
+                        if set(class_muts[md["mutation_type"]]) == set("non_neutral"):
+                            non_neu_positions.append(m.site)
+                site_class[j] = class_muts[mut_hits[0]] if len(
+                    mut_hits) == 1 else "double_hit"
+            assert sum(site_class == None) == 0
```

Comment:
Can we reuse the codes `_generate_fs_from_ts` in `ts2fs.py`?

Reply:
there's probably a way to do this but it would take a bit of coding as currently `_generate_fs_from_ts` returns an array of allele frequency spectra for each class of mutation. this could be a good thing to wrap in a function however, and would be a simple enough PR